src/Debugger/Impl/DebugEvaluationResult.cs

```diff
@@ -2,113 +2,114 @@
-        EnvName = 1 << 14,
-        Flags = 1 << 15,
-        Children = Expression | Length | AttrCount | SlotCount | NameCount | Flags,
-    }
-
-    internal static class DebugEvaluationResultFieldsExtensions {
-        private static readonly Dictionary<DebugEvaluationResultFields, string> _mapping = new Dictionary<DebugEvaluationResultFields, string> {
-            [DebugEvaluationResultFields.Expression] = "expression",
-            [DebugEvaluationResultFields.Kind] = "kind",
-            [DebugEvaluationResultFields.Repr] = "repr",
-            [DebugEvaluationResultFields.ReprDeparse] = "repr.deparse",
-            [DebugEvaluationResultFields.ReprToString] = "repr.toString",
-            [DebugEvaluationResultFields.ReprStr] = "repr.str",
-            [DebugEvaluationResultFields.TypeName] = "type",
-            [DebugEvaluationResultFields.Classes] = "classes",
-            [DebugEvaluationResultFields.Length] = "length",
-            [DebugEvaluationResultFields.SlotCount] = "slot_count",
-            [DebugEvaluationResultFields.AttrCount] = "attr_count",
-            [DebugEvaluationResultFields.NameCount] = "name_count",
-            [DebugEvaluationResultFields.Dim] = "dim",
-            [DebugEvaluationResultFields.EnvName] = "env_name",
-            [DebugEvaluationResultFields.Flags] = "flags",
-        };
+    /// <summary>
+    /// Describes the result of evaluating an R expression, with additional metadata such as type information.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// All properties of this class, and its derived classes such as <see cref="DebugValueEvaluationResult"/>, implement snapshot
+    /// semantics - that is, they have values that describe the result of evaluation at the point where it took place, and do not
+    /// reflect any changes to R state that have occurred since then. For methods, however, this can vary; if the method does not
+    /// implement snapshot semantics, then its documentation will reflect that.
+    /// </para>
+    /// <para>
+    /// Those methods that do not have snapshot semantics will return fresh results obtained by evaluating the expression that produced
+    /// this result in its original context. If the context is no longer available (for example, it was a stack frame that is no
+    /// longer there), the results are undefined. 
+    /// </remarks>
+    public abstract class DebugEvaluationResult {
+        public DebugSession Session { get; }
 
-        public static string ToRVector(this DebugEvaluationResultFields fields) {
-            var fieldNames = _mapping.Where(kv => fields.HasFlag(kv.Key)).Select(kv => "'" + kv.Value + "'");
-            return Invariant($"base::c({string.Join(", ", fieldNames)})");
-        }
-    }
+        /// <summary>
+        /// R expression designating the environment in which the evaluation that produced this result took place.
+        /// </summary>
+        public string EnvironmentExpression { get; }
```

Comment:
No-one uses it outside of the class (but I saw no reason to make it private). Inside the class, the first overload of `EvaluateAsync` uses it, as do `GetChildrenAsync` in `DebugValueEvaluationResult`.